### DISCUSSION TITLE
API Keys with Group Permissions

### DISCUSSION THREAD

**Discussion by aaniceto** [requarks/wiki#4753] [Help / Questions] [answered]
Hi,

We use wikijs 2.5.219 and we have different group permissions that we assign to different users.

We need to be able to create an API Key with read-only permissions to pages/etc

When we try to create a new API Key the option Full Access is selected, and can't be unselected. And the dropdown to choose a Permission Group is disabled.

![image](https://user-images.githubusercontent.com/19576660/144702746-520ad111-5454-46d9-9f59-fece77e4dde4.png)

Is Group Permissions working in API Keys? I assume yes because the dropdown is there.

Thanks in advance,
Alexandre

**Comment by tim-elmer** [comment]
Was hoping to use GraphQL to add a query-based component to a wiki page, but the requirement of full API access makes this a bit of a non-starter. I don't quite feel like spinning up an intermediate server to handle this one feature, so being able to trust a limited-scope key in a `fetch` call would be wonderful.

**Comment by NGPixel** [comment, marked as answer]
You need to create a new group first. Then you'll be able to deselect "Full Access" and select that group. The reason you can't deselect it by default is because you can't choose the Guest group and the Administrators group is the same as Full Access.

**Reply by tim-elmer** [reply]
Any additional constraints on those groups? I have 6 non-default groups and I cannot uncheck the "Full Access" checkbox. The "Group" select box appears to be disabled.

**Reply by tim-elmer** [reply]
Nevermind, apparently the checkbox does not work, but its label does.

**Reply by NGPixel** [reply]
Good catch, that's a bug. I'll push a fix for the next update.
